mathesar/utils/schemas.py

```diff
@@ -31,6 +31,6 @@ def reflect_schemas_from_database(database):
         for oid in db_schema_oids
     ]
     for schema in Schema.objects.all():
-        if schema.oid not in db_schema_oids:
```

Comment:
This makes me realize that we'll probably have a problem if we remove a database.  Wouldn't those schemas just hang around indefinitely, attached to nothing?

Reply:
We could add a `deleted` field to schemas (and databases) and mark them as deleted if we don't see them connected.

Reply:
Could we solve it by referencing the database in a foreign key from the Schema model, and using `on_delete=models.CASCADE`?  That way, if we notice a DB is missing on reflection, we can delete the DB entry and let it cascade to the schema.  I'd rather avoid building up "deleted" resources in the models tables.

Reply:
that makes sense. I think it would make sense to have a persistent model for the DBs, especially given the new design spec where we show databases that we don't have a connection to with an error message. I think we should mark the DB as deleted rather than deleting the model, though.

Reply:
I think this makes sense to do in https://github.com/centerofci/mathesar/issues/311, not here.